receiver/pluginreceiver/supplied_plugins_test.go

```diff
@@ -41,17 +63,46 @@ func TestValidateSuppliedPlugins(t *testing.T) {
 		t.Run(fmt.Sprintf("Loading %s", entry.Name()), func(t *testing.T) {
 			t.Parallel()
 			fullFilePath, err := filepath.Abs(filepath.Join(pluginDirPath, entryName))
-			assert.NoError(t, err, "Failed to determine path of file %s", entryName)
+			require.NoError(t, err, "Failed to determine path of file %s", entryName)
 
 			// Load the plugin
 			plugin, err := LoadPlugin(fullFilePath)
-			assert.NoError(t, err, "Failed to load file %s", entryName)
+			require.NoError(t, err, "Failed to load file %s", entryName)
 
 			cfg, err := config.NewComponentIDFromString("test")
 			require.NoError(t, err)
 
-			_, err = plugin.Render(map[string]any{}, cfg)
-			assert.NoError(t, err, "Failed to render config for plugin %s", entryName)
+			// Render the config
+			renderedCfg, err := plugin.Render(map[string]any{}, cfg)
+			require.NoError(t, err, "Failed to render config for plugin %s", entryName)
+
+			// Check receivers and filter out checking those that are not supported by the OS
+			for id := range renderedCfg.Receivers {
+				componentID, err := config.NewComponentIDFromString(id)
+				require.NoError(t, err, "Failed to parse component ID %s for plugin %s", id, entryName)
+
+				switch componentID.Type() {
+				case windowseventlogreceiver.NewFactory().Type():
+					if runtime.GOOS != "windows" {
+						return
+					}
+				case journaldreceiver.NewFactory().Type():
+					if runtime.GOOS != "linux" {
+						return
+					}
```

Comment:
Should there be `darwin` case or a default case triggering an error?

Reply:
I don't think so. The two components that are called out on work on specific operating systems. There is no receiver that only works on mac. Though I think syslog won't work on windows, will need to add that in.